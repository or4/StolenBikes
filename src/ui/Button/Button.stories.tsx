import { storiesOf } from '@storybook/react';
import * as React from 'react';
import { Button } from './Button';

storiesOf('Button', module).add('with text', () => <Button text={'123'} onClick={() => {}} />);
