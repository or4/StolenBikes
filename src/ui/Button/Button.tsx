import React from 'react';
import { space, width, fontSize, color, WidthProps, SpaceProps } from 'styled-system';
import styled, { ThemeProvider } from 'styled-components';
import styledTs from 'styled-components-ts';
import { theme } from '../theme';

type StyledButtonProps = WidthProps & SpaceProps;

export const StyledButton = styledTs<StyledButtonProps>(styled.div)`
	${space}
	${width}
	${fontSize}
	${color}
`;

StyledButton.defaultProps = {
    p: 2,
    color: 'orange',
};

interface Props {
    text: string;
    onClick: () => void;
}

interface State {
    test?: boolean;
}

export class Button extends React.PureComponent<Props, State> {
    public state: State = {};
    public componentDidMount() {
        this.setState({ test: true });
    }
    public render() {
        return (
            <ThemeProvider theme={theme}>
                <StyledButton>{this.props.text}</StyledButton>
            </ThemeProvider>
        );
    }
}
