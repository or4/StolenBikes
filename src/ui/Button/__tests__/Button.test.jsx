import React from 'react';
import { shallow, render, mount } from 'enzyme';
import { Button } from '../Button';

describe('MyComponent', () => {
    it('should render correctly in "debug" mode', () => {
        const button = <Button text={'test buttons'} onClick={() => {}} debug />;

        expect(shallow(button)).toMatchSnapshot();
        expect(render(button)).toMatchSnapshot();

        const mounted = mount(button);
        expect(mounted).toMatchSnapshot();
        mounted.unmount();

        const clickFn = jest.fn();

        const component = shallow(<Button onClick={clickFn} />);
        component.find('#my-button-two').simulate('click');
        expect(clickFn).toHaveBeenCalled();
    });
});
