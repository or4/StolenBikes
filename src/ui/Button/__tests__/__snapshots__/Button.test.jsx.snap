// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly in "debug" mode 1`] = `
<styled.div
  id="my-button-two"
  onClick={[Function]}
>
  Button 
  true
</styled.div>
`;

exports[`MyComponent should render correctly in "debug" mode 2`] = `
<div
  class="sc-bdVaJa ipstQh"
  id="my-button-two"
>
  Button undefined
</div>
`;

exports[`MyComponent should render correctly in "debug" mode 3`] = `
<Button
  debug={true}
  onClick={[Function]}
  text="test buttons"
>
  <styled.div
    id="my-button-two"
    onClick={[Function]}
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "ipstQh",
            "rules": Array [
              "
    display: inline-block;
    border: 1px solid black;
    padding: 5px;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      id="my-button-two"
      onClick={[Function]}
    >
      <div
        className="sc-bdVaJa ipstQh"
        id="my-button-two"
        onClick={[Function]}
      >
        Button 
        true
      </div>
    </StyledComponent>
  </styled.div>
</Button>
`;
